prql/src/semantic/materializer.rs

```diff
@@ -357,10 +419,10 @@ aggregate [
         let ast = parse(
             r#"
 func lag_day x = s"lag_day_todo({x})"
-func ret x = x / (lag_day x) - 1 + dividend_return
+func ret x dividend_return = x / (lag_day x) - 1 + dividend_return
```

Comment:
I don't quite understand this change — `dividend_return` was supposed to be a column, rather than a param. Tell me if I'm missing something

Reply:
Functions cannot reference columns, because they could be used in different context, where that column would not be available.

Eg:
```
from employees
select first_name
derive [a: ret 15]   # there is only first_name in scope here, cannot reference dividend_return!
```

Reply:
What do you think about "late binding" them? i.e. as though someone copy-pasted the function definition. In the way I was thinking about it, having:

```elm
from a
select (b / (lag_day b) - 1 + dividend_return)
```

..._would_ be allowed, and so having `ret b` would be equivalent to that.

I had thought of this as quite an important feature — otherwise functions can't be used to do things like "add X and subtract Y and multiply by Z" without specifying X & Y & Z each time.

I recognize that this sort of late-binding isn't common in languages; it's more like a macro. But also we're using `dividend_return` as more like another language would use `"dividend_return"` — i.e. if it's not specified as a parameter, then it's a column name, and can be treated as a literal.

Does my point make sense? WDYT?

Reply:
Our functions are very similar to macros, which is a case for this feature. I see the use case.

But, this kinda uncommon for other languages... I'm hesitant about this mostly because it breaks current resolver/materialiazer architecture. But to be honest, this would be useful to have.

Currently it can be bypassed by using s-strings, though.

Reply:
Yes true re s-strings. 

Is it OK if I add an issue so we can track it?